src/types.ts

```diff
@@ -15,7 +15,11 @@ export type StateMachineContextValue = {
   setState: React.Dispatch<React.SetStateAction<GlobalState>>
 };
 
-export type MiddleWare = <T>(arg: T) => T;
+export type MiddleWare = (
+  state: GlobalState,
+  callbackName: string,
+  payload: any,
```

Comment:
maybe switch the order? I think fewer people would use the `callbackName`?

Reply:
Done with `feat: Change order of middleware arguments` ce8f6d5